<?xml version="1.0"?>

<launch>
    <arg name="mapping" default="true" />
    <arg name="tracking" default="true" />
    <arg name="adjust_throttle" default="false" />
    <arg name="lidar" default="false" />
    <arg name="model" default="true" />
    <arg name="real" default="false" />
    <arg name="mixed" default="true" />
    <arg name="sim" default="false" />


    <rosparam command="load" file="$(find mixed_reality)/launch/params.yaml" />
    <param name="mapping" value="$(arg mapping)" />
    <param name="tracking" value="$(arg tracking)" />
    <param name="real" value="$(arg real)" />
    <param name="mixed" value="$(arg mixed)" />
    <param name="sim" value="$(arg sim)" />
    

    <node name="sim_node" pkg="mixed_reality" type="simulator_node.py" cwd="node" required="true"/>
    <!--<node name="control_node" pkg="mixed_reality" type="control_node.py" cwd="node"/>-->
    <node name="keyboard_node" pkg="mixed_reality" type="keyboard_node.py" cwd="node" required="true" output="screen"/>
    <group if="$(arg tracking)">
        <node name="tracking_node" pkg="mixed_reality" type="vicon_tracking.py" cwd="node"/>
        <node name="car_node" pkg="mixed_reality" type="car_node.py" cwd="node"/>
    </group>
    <group if="$(arg adjust_throttle)">
        <node name="throttle_sim" pkg="mixed_reality" type="throttle_sim.py" cwd="node"/>
        <group if="$(arg tracking)">
            <node name="throttle_car" pkg="mixed_reality" type="throttle_car.py" cwd="node"/>
        </group>
    </group>
    <group if="$(arg lidar)">
        <node name="lidar_depth_node" pkg="mixed_reality" type="lidar_depth_node.py" cwd="node"/>
        <node name="pointcloud_node" pkg="mixed_reality" type="pointcloud_node.py" cwd="node"/>
    </group>
    <group if="$(arg model)">
        <node name="model_node" pkg="mixed_reality" type="model_node.py" cwd="node" required="true"/>
        <group unless="$(arg sim)">
            <node name="car_camera" pkg="mixed_reality" type="camera_node.py" cwd="node"/>
        </group>
        <group unless="$(arg real)">
            <group unless="$(arg sim)">
                <node name="mixer_node" pkg="mixed_reality" type="mixer_node.py" cwd="node"/>
            </group> 
        </group>
    </group>

</launch>